### PR TITLE
setup even handlers for buttons

### PR DESCRIPTION
**Pull request by Reggeditt**
- Installed [big.js](https://www.npmjs.com/package/big.js) using NPM
- Analyzed the files calculate.js and operate.js, understood what they do,
 then Imported the files in your Calculator.js component
- Implemented click event handlers using the math logic from calculate.js and operate.js so that all calculator buttons work according to their functionality.

### CODE REVIEW

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Ghiftee) [inline src/logic/operate.js]: - [ ] Kindly do not make any changes to this file. Please do not disable linters as they help us write cleaner code.

File: src/componets/calculator.js [hunk 2]
```diff
@@ -1,29 +1,37 @@
...
+import calculate from '../logic/calculate';
 
 function Calculator() {
+  const display = useRef(null);
+  let result = { total: null, next: null, operation: null };
+  const getButtonName = (e) => {
+    const buttonName = e.target.textContent;
+    result = calculate(result, buttonName);
+    display.current.textContent = result.next || result.operation || result.total || '0';
+  };
```
Comment (Ghiftee) [inline src/componets/calculator.js]: - [ ] Kindly use `useState` to handle state here and not `useRef` as you did. `useState` is designed specifically for managing state in React components while `useRef` is typically used for accessing and manipulating DOM elements.

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @Reggeditt 👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ App runs smoothly
✔️ Logic files are added
✔️ README is professional

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/logic/operate.js (hunk 1 above)
Comment (Reggeditt) [reply src/logic/operate.js]: Do you have any suggestions for me about how to solve the problem? I have tried everything and I still get the error with the import statement. @Ghiftee . thanks

Comment (Ghiftee) [reply src/logic/operate.js]: Hi @Reggeditt please try to install `big.js` as a dependency then if you use npm for the installation, you can require like this `const Big = require('big.js');` It should fix the error

File: src/componets/calculator.js (hunk 2 above)
Comment (Reggeditt) [reply src/componets/calculator.js]: fixed

File: src/logic/operate.js (hunk 1 above)
Comment (Reggeditt) [reply src/logic/operate.js]: I tried that. It didnt work. And I also checked the dependencies, big.js is listed there so I don't know why the error is showing up in the first place.
When I use const Big = require('big.js');, it gives me the same error.
@Ghiftee

Comment (Reggeditt) [reply src/logic/operate.js]: Please advice 🙏🏾 @Ghiftee

Comment (Reggeditt) [reply src/logic/operate.js]: let me show you..

**Review by DammyShittu — APPROVED** [review]
Hi @Reggeditt,

## APPROVED 🟢

## Highlights
✔️ useState in calculator

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by wayungi — COMMENTED** [review]
Hi @Reggeditt , 
image

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ❌ ❌ 

- [ ] You have submitted an approved PR for a new milestone. Kindly create a new PR for every milestone of the project

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**